### PR TITLE
Extract user deletion into a new service

### PR DESCRIPTION
**Pull request by FCsongradi**
Pull request for [issue 915](https://github.com/apigee/apigee-edge-drupal/issues/915).

### CODE REVIEW

File: src/UserRemovalHandler.php
Comment (mxr576) [inline src/UserRemovalHandler.php:35]: Suggedted name `RemoveDeveloperWithUserUserRemovalHandler`.

Make  the class `final` mark the `entityTypeManager` property as `private`, and use native typehints.

Comment (mxr576) [inline src/UserRemovalHandler.php]:
```suggestion
final class RemoveDeveloperWithUserUserRemovalHandler implements UserRemovalHandlerInterface {
```

File: src/UserRemovalHandler.php
Comment (mxr576) [inline src/UserRemovalHandler.php]:
```suggestion
  private EntityTypeManagerInterface $entityTypeManager;
```

File: src/UserRemovalHandler.php
Comment (mxr576) [inline src/UserRemovalHandler.php]:
```suggestion
   * RemoveDeveloperWithUserUserRemovalHandler constructor.
```

File: src/UserRemovalHandler.php
Comment (mxr576) [inline src/UserRemovalHandler.php]:
```suggestion
  public function __invoke(UserInterface $account): void {
```

File: src/UserRemovalHandler.php
Comment (mxr576) [inline src/UserRemovalHandler.php]: Would be nice to have an `info` level log message in logs about the removal of the developer. I know this is a change compared with the original implementation, but only a minor one.

File: src/UserRemovalHandlerInterface.php
Comment (mxr576) [inline src/UserRemovalHandlerInterface.php]:
```suggestion
  public function __invoke(UserInterface $account): void;
```

File: apigee_edge.module
Comment (mxr576) [inline apigee_edge.module]:
```suggestion
  (\Drupal::service('apigee_edge.user_removal_handler'))($account);
```

File: apigee_edge.module
Comment (mxr576) [inline apigee_edge.module]: Maybe place this class and the interface under the `Drupal\apigee_edge\User` namespace

File: src/UserRemovalHandler.php
Comment (FCsongradi) [reply src/UserRemovalHandler.php]: Added info level logging when the developer is deleted.

File: apigee_edge.module
Comment (FCsongradi) [reply apigee_edge.module]: Refactored

**Comment by mxr576** [comment]
This is a 2.x backport of #919.

**Review by mxr576 — APPROVED** [review]

**Comment by mxr576** [comment]
@FCsongradi #919 has changes, backporting them can wait until a review from maintainers.

**Comment by mxr576** [comment]
The build failed because Codecove update failed.

Should be ready for review.

**Comment by shishir-intelli** [comment]
> The build failed because Codecove update failed.
> 
> Should be ready for review.

@mxr576 
Lets rerun it, probably codecov uploading issue.

**Review by shishir-intelli — APPROVED** [review]
LGTM

**Pull request merged**
